prime-router/src/main/kotlin/azure/BlobAccess.kt

```diff
@@ -193,6 +193,36 @@ class BlobAccess() : Logging {
             return getBlobClient(blobUrl, blobConnInfo).exists()
         }
 
+        /**
+         * Copies all blobs prefixed with the [directory] value from the soure to destination
+         *
+         * @param directory - the prefix (or directory) containing the blobs to be copied
+         * @param source - the source account to copy from
+         * @param destination - the destination to copy to
+         */
+        fun copyDir(directory: String, source: BlobContainerMetadata, destination: BlobContainerMetadata) {
```

Comment:
I mentioned this on your other PR but not sure it got seen. All the operations in this file are pretty heavy. Have we thought about making them async?

Reply:
As in making this `suspendable`?  I don't see why we couldn't... though I think the first use case for this won't take advantage of it

Reply:
Not a blocker for merging since this is the pattern we've been using.

It might be a good candidate for refactoring in the future if we run into performance issues. Moving IO functionality into suspended functions is generally good practice.